### DISCUSSION TITLE
[SO 2.3.70 AWS AMI] Zeek/Suricata/Strelka: File extraction does not seem to be working

### DISCUSSION THREAD

**Discussion by jeFF0Falltrades** [Security-Onion-Solutions/securityonion#5547] [Unsupported Versions]
HI all,

I am working with a fresh install of the SO 2.3.70 AMI in AWS.

I have traffic mirrored from a Windows server that I am running simulations on to the sniffing interface of SO, and for some reason, I cannot get any files to show up in Strelka in the "Hunt" or "PCAP" view or on disk in `/nsm/strelka/processed` or `unprocessed` or `staging`, nor in `/nsm/zeek/extracted/complete`.

I can see the transfer of the file using both Zeek and Suricata for metadata extraction in my network in the`files` logs, but for some reason I cannot get any PCAP to show me the traffic where the file was transferred, nor can I get anything out of Strelka.

Things I have tried so far to no avail:

* Restarting the server
* Switching `mdengine` in `global.sls` to `SURICATA` and then back to `ZEEK` to see if using Suricata would make a difference (it did not)
* Restarted strelka, zeek, suricata, and pcap
* Reinstalled SO from scratch
* Checked logs for strelka (no log entries at all are present in `strelka.log`), zeek (shows file was identified but nothing else), suricata (same as zeek), and PCAP 
* Cleared out all PCAP data

Not sure if related, but the odd thing about the PCAP view is I can set up a filter with no SRC/DEST IP specified, and I will see PCAP traffic, but as soon as I try to set up a filter for a SRC/DEST IP, it shows that there is "No data available".

Output from Zeek `files.log` showing recognition of file transfer in network traffic:

```
{
  "ts": 1631816879.389073,
  "fuid": "FBCZRJxD1qp77FCTh",
  "tx_hosts": ["192.168.1.50"],
  "rx_hosts": ["192.168.1.201"],
  "conn_uids": ["CTx3XB2FTsNgfnCC29"],
  "source": "HTTP",
  "depth": 0,
  "analyzers": ["MD5", "PE", "EXTRACT", "SHA1"],
  "mime_type": "application/x-dosexec",
  "filename": "REDACTED.exe",
  "duration": 0.0000171661376953125,
  "local_orig": true,
  "is_orig": true,
  "seen_bytes": 6144,
  "missing_bytes": 0,
  "overflow_bytes": 0,
  "timedout": false,
  "md5": "REDACTED",
  "sha1": "REDACTED",
  "extracted": "HTTP-FBCZRJxD1qp77FCTh.exe",
  "extracted_cutoff": false
}
```

Entry showing transfer of file:

![image](https://user-images.githubusercontent.com/8444166/133670670-53db3b2a-a729-46fd-834e-110fcfbcc867.png)

But no matching PCAP when clicking on the ID of that entry:

![image](https://user-images.githubusercontent.com/8444166/133671078-c8484863-685c-4ca8-9c1e-ff5d4994f75f.png)

However, there is at least some PCAP data flowing in, because when I set a filter with no src/dest IP, I see results:

![image](https://user-images.githubusercontent.com/8444166/133671565-df39d72d-6c68-4c6f-8aad-5a68967aa8b3.png)

Any help is greatly appreciated - just don't know where to even start looking for a disconnect.

EDIT: I should add that manually copying the file to `/nsm/strelka/unprocessed` locally works fine and it does show up then; It's just not extracting from web traffic for some reason.

**Comment by dougburks** [comment]
I think this is related to the fact that AWS encapsulates their traffic in VXLAN.

**Reply by jeFF0Falltrades** [reply]
Thanks for the response @dougburks ; I’m not sure how this would impact file extraction though as the versions of Zeek/Suricata included with 2.3 are [capable of parsing VXLAN traffic](https://docs.securityonion.net/en/2.3/cloud-ami.html) and I can see network traffic and alerts being generated (just seemingly not *all* of the packets are being captured in the PCAP view). 

I verified that my traffic mirror also has all protocols being mirrored.

**Reply by dougburks** [reply]
Is it possible you are running into this Strelka issue?
https://github.com/Security-Onion-Solutions/securityonion/issues/5578

**Reply by jeFF0Falltrades** [reply]
Thank you again for following up, Doug!

I should have mentioned that during troubleshooting I did actually run into that issue with YARA when testing with my manually copied files into the Strelka “unprocessed” folder. 

I blew away all YARA rules, which fixed that issue, but unfortunately did not make any progress on this issue. 

The problem here seems to be that files are not getting to Strelka in the first place. 

I will continue testing and try to figure out where exactly the disconnect is. 

As you see in the output from Zeek above, it looks like Zeek even recognizes the file and the “EXTRACT” analyzer is listed, but for some reason I never see a file created/saved on the SO server in any Zeek/Suricata extracted folder or Strelka “unprocessed” folder.

**Reply by Tedus007** [reply]
Hello Guys, 

I believe I'm facing the same problem as well. I tried to download a .doc file via http, but nothing appears in Strelka, nor zeek logs.

The file isn't extracted.

**Reply by jeFF0Falltrades** [reply]
@Tedus007 Does it show up in traffic logs at least?

I ask because it seems like (as seen above) I can see logs under “files” and “http” that show the file was requested, but after that nothing appears to happen. 

I’d be curious to know if you see the same, or if you don’t seen anything at all.

**Reply by Tedus007** [reply]
@jeFF0Falltrades 

I can see traffic logs.
However I noticed one meta showing bytes missing file.bytes.missing


**file.bytes.missing | 191,820
  | file.bytes.overflow | 0
  | file.bytes.seen | 206,417
  | file.bytes.total | 398,237**

I don't understand that part does this mean Zeek didn't get the whole session and drops occurred this is why file wasn't extracted?

But it detects the whole file size correct!!

**Reply by jeFF0Falltrades** [reply]
Yes @Tedus007 , unfortunately in this case it looks like the Issue you are having might be distinct from this one - yours may be related to packet loss. 

To the point about Zeek knowing the full file size, depending on how the file was requested, sometimes the total size of a file is sent before the download/transfer takes place (such as in the content-length field for an HTTP request). This allows an analyzer like Zeek to know the full size to expect, and to detect when packet loss occurs.
